pkg/api/server/v1alpha2/log/gcs_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package log
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-replayers/httpreplay"
+	hrgoog "github.com/google/go-replayers/httpreplay/google"
+	server "github.com/tektoncd/results/pkg/api/server/config"
+	"gocloud.dev/gcp"
+	"google.golang.org/api/option"
+)
+
+var (
+	// Record is true if the tests are being run in "record" mode.
+	// If replay file needs to be generated then please give -record during run of tests
+	Record = flag.Bool("record", false,
+		"whether to run tests against cloud resources and record the interactions")
```

Comment:
Do we intend to use this flag with other cloud provider tests?

Reply:
Yes. When I add other storage driver, I would go with record and replay approach for unit testing.

Reply:
Perhaps add a TODO comment to move this flag to a more central location as other cloud providers are added,

Reply:
Running the tests with record flag results an error:
```
[ results] -> go test -count=1 ./pkg/api/server/v1alpha2/log/ -record
--- FAIL: TestGCSReadFrom (0.14s)
    gcs_test.go:53: Recording GCS API responses into replay file testdata/TestGCSReadFrom.replay
    gcs_test.go:107: failed to get default credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
--- FAIL: TestGCSWriteTo (0.00s)
    gcs_test.go:53: Recording GCS API responses into replay file testdata/TestGCSWriteTo.replay
    gcs_test.go:107: failed to get default credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
--- FAIL: TestGCSDelete (0.00s)
    gcs_test.go:53: Recording GCS API responses into replay file testdata/TestGCSDelete.replay
    gcs_test.go:107: failed to get default credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
FAIL
FAIL    github.com/tektoncd/results/pkg/api/server/v1alpha2/log 0.146s
FAIL
```

Reply:
@enarha You need to supply a GCP cred in environment variable: 

```export GOOGLE_APPLICATION_CREDENTIALS=/path-to/creds.json``

Reply:
Indeed, that worked. The reason I tried though is that I wanted to confirm where the .replay files are stored and they override the `pkg/api/server/v1alpha2/log/testdata/{TestGCSDelete,TestGCSReadFrom,TestGCSWriteTo}.replay` files which is not ideal IMO. Tests should not update the repo files. Can we change that to some temp location (/tmp)?

Reply:
Like the tests that run for this pr, it needed a replay file.  `-record` is needed only if we need to update replay files due to some driver change or Google API changes which will be quite rare.

Reply:
OK. Fine with me. Maybe just a comment that the `-record` should be used when we need to update the replay files due to API changes. Feel free to resolve the thread.